modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -4631,6 +4631,115 @@ CV_EXPORTS_W double getFontScaleFromHeight(const int fontFace,
                                            const int pixelHeight,
                                            const int thickness = 1);
 
+/** @brief Wrapper on top of a truetype/opentype/etc font, i.e. Freetype's FT_Face.
+
+The class is used to store the loaded fonts;
+the font can then be passed to the functions
+putText and getTextSize.
+*/
+class CV_EXPORTS_W_SIMPLE FontFace
+{
+public:
+    /** @brief loads default font */
+    CV_WRAP FontFace();
+    /** @brief
+       loads font at the specified path or with specified name.
+       Empty fontPathOrName means the default embedded font.
+    */
+    CV_WRAP FontFace(const String& fontPathOrName);
+
+    ~FontFace();
+
+    /** @brief loads new font face */
+    CV_WRAP bool set(const String& fontPathOrName);
+    CV_WRAP String getName() const;
+
+    /** @brief sets the current variable font instance.
+        @param params The list of pairs key1, value1, key2, value2, ..., e.g.
+             `myfont.setInstance({CV_FOURCC('w','g','h','t'), 400<<16, CV_FOURCC('s','l','n','t'), -(15<<16)});`
+        Note that the parameter values are specified in 16.16 fixed-point format, that is, integer values
+        need to be shifted by 16 (or multiplied by 65536).
+    */
+    CV_WRAP bool setInstance(const std::vector<int>& params);
+    CV_WRAP bool getInstance(CV_OUT std::vector<int>& params) const;
```

Comment:
"Instance" of what?
`Property` is less confused name.

> CV_FOURCC('s','l','n','t')

Public API must be simple and readable. Obfuscation should not pass.

---

What is the problem to add `setWeight()/getWeight()`, etc?

Reply:
each variable font defines its own set of so-called "axes"; they all form a point in the font's "design space", or "instance" in other words. This API gives user full flexibility to set any parameters of any variable font he or she is using. The parameters should be set at once, otherwise it's not clear whether we want to start from scratch or just modify individual axes.

Reply:
I disagree; please, put it in as-is